library/Vanilla/OpenAPIBuilder.php

```diff
@@ -81,9 +87,31 @@ public function getFullOpenAPI(): array {
         }
 
         $result = require $this->cachePath;
+
+        // Reapply URL even after pulling from cache.
+        // A site may be accessed from multiple URLs and share the same cache.
+        $result = $this->applyCorrectApiBaseUrl($result);
```

Comment:
I think naming convention should be to avoid `Correct` or any other unrelated adjectives to be in method name.
Such kind of names bring the idea that there is `applyIncoorectApiBaseUrl` or `applyWrongApiBaseUrl` somewhere else. Or I could be wrong and we have theme elsewhere?
```suggestion
        $result = $this->applyApiBaseUrl($result);
```

Reply:
We don't necessarily have a method for them, but the default was certainly incorrect.

I've renamed it to `applyRequestBasedApiBasePath()`